drivers/gpio/gpio_mcp23s17.h

```diff
@@ -55,6 +55,7 @@ struct mcp23s17_config {
 	const uint32_t freq;
 	const char *const cs_dev;
 	const uint32_t cs_pin;
+	const uint8_t cs_flags;
```

Comment:
Since the structure is being modified it would be nice to adjust these to use the (new-ish) canonical types `gpio_pin_t` and `gpio_dt_flags_t` (after confirming that usage doesn't require `gpio_flags_t`).  But I'd accept an "out of scope" judgment from the contributor.

Reply:
When going through the various drivers, I think I found 4 different ways that the SPI buses are handled through configuration. I would prefer to leave that out of this pull request, but wholeheartedly support a cleanup effort.

Reply:
Actually, your comments make me think there is an actual issue in these drivers: it would be worth modifying these so instead of setting custom attributes in a custom *_config structure, that they would directly initialize their dedicated spi_cs_control attribute in their private data. That would lower the size of their *_config structure, and avoid unnecessary copy of these values.

That would be for another PR, nothing to be fixed here.